notebook/agentchat_oai_code_interpreter.ipynb

```diff
@@ -0,0 +1,161 @@
+    "```python\n",
+    "config_list = [\n",
+    "    {\n",
+    "        \"model\": \"gpt-4\",\n",
+    "        \"api_key\": \"<your OpenAI API key>\",\n",
+    "    },  # OpenAI API endpoint for gpt-4\n",
+    "    {\n",
+    "        \"model\": \"gpt-35-turbo-0631\",  # 0631 or newer is needed to use functions\n",
+    "        \"base_url\": \"<your Azure OpenAI API base>\", \n",
+    "        \"api_type\": \"azure\", \n",
+    "        \"api_version\": \"2023-08-01-preview\", # 2023-07-01-preview or newer is needed to use functions\n",
+    "        \"api_key\": \"<your Azure OpenAI API key>\"\n",
+    "    }\n",
+    "]\n",
+    "```\n",
+    "\n",
+    "You can set the value of config_list in any way you prefer. Please refer to this [notebook](https://github.com/microsoft/autogen/blob/main/notebook/oai_openai_utils.ipynb) for full code examples of the different methods."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Perform Tasks Using Code Interpreter\n",
+    "\n",
+    "In this example, we demonstrate task solving using `GPTAssistantAgent` with code interpreter. Pass `code_interpreter` in `tools` parameter to enable `GPTAssistantAgent` with code interpreter. It will write code and automatically execute it in a sandbox. The agent will receive the results from the sandbox environment and act accordingly.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[33muser_proxy\u001b[0m (to Coder Assistant):\n",
+      "\n",
+      "If $725x + 727y = 1500$ and $729x+ 731y = 1508$, what is the value of $x - y$ ?\n",
+      "\n",
+      "--------------------------------------------------------------------------------\n",
+      "\u001b[33mCoder Assistant\u001b[0m (to user_proxy):\n",
+      "\n",
+      "The value of \\( x - y \\) is \\(-48\\). \n",
+      "\n",
+      "TERMINATE\n",
+      "\n",
+      "\n",
+      "--------------------------------------------------------------------------------\n"
+     ]
+    }
+   ],
+   "source": [
+    "from autogen.agentchat.contrib.gpt_assistant_agent import GPTAssistantAgent\n",
+    "from autogen.agentchat import AssistantAgent, UserProxyAgent\n",
+    "\n",
+    "# Initiate an agent equipped with code interpreter\n",
+    "gpt_assistant = GPTAssistantAgent(\n",
+    "    name=\"Coder Assistant\",\n",
+    "    llm_config={\n",
+    "        \"tools\": [\n",
+    "            {\n",
+    "                \"type\": \"code_interpreter\"\n",
+    "            }\n",
+    "        ],\n",
+    "        \"config_list\": config_list,\n",
+    "    },\n",
+    "    instructions=\"You are an expert at solving math questions. Write code and run it to solve math problems. Reply TERMINATE when the task is solved and there is no problem.\",\n",
+    ")\n",
+    "\n",
+    "user_proxy = UserProxyAgent(\n",
```

Comment:
Do we need to disable the code execution? From @gagb 's experience, sometimes the code interpreter will fail and a python code will still be suggested. So that could be a reason not to disable the code execution.

Reply:
Sounds great. I readded `code_execution_config`.